File: 71-simplify-path/simplify-path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        stringstream ss(path);
        string ans ="";
        string temp;
        stack<string> s;
        while(getline(ss,temp,'/')){
            
            if(temp =="." || temp=="") continue;
            else if(temp==".."){
              if(!s.empty()) s.pop();
            }
            else{
                s.push(temp);
            }
        }
        while(!s.empty()){
            
        ans = "/" + s.top() + ans;
        s.pop();
        }
        if(ans=="") return "/";
        return ans;
        
    }
};

File: 71-simplify-path/simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] paths = path.split("/");
        Stack<String> st = new Stack<>();
        for(String s : paths){
            if(s.equals(".") || s.equals("")) continue;
            else if(s.equals("..")){
                if(!st.isEmpty()) st.pop();
            }
            else st.add(s);
        }
        StringBuilder sb = new StringBuilder("");
        if(st.isEmpty()) return "/";
        for(String i : st){
            sb.append("/").append(i);
        }
        return sb.toString();
    }
}